modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/extra/remote-app/src/routes/MDFClaimForm/components/MDFClaimPage/schema/yup.ts

```diff
@@ -169,11 +141,138 @@ const claimSchema = object({
 								}
 							),
 				}),
-				metrics: string().max(
-					350,
-					'You have exceeded the character limit'
+				listOfQualifiedLeads: mixed()
+					.when('selected', {
+						is: (selected: boolean) => selected,
+						then: (schema) =>
+							schema
+								.test(
+									'fileSize',
+									validateDocument.fileSize.message,
+									(listOfQualifiedLeads) => {
+										if (
+											listOfQualifiedLeads &&
+											!listOfQualifiedLeads.id
+										) {
+											return listOfQualifiedLeads
+												? Math.ceil(
+														listOfQualifiedLeads.size /
+															1000
+												  ) <=
+														validateDocument
+															.fileSize.maxSize
+												: false;
+										}
+
+										return true;
+									}
```

Comment:
Improve this logic, the ternary it's not necessary since the same condition is validate in if statement